File: 2200_FindAllK-DistantIndicesinanArray/2200_FindAllK-DistantIndicesinanArray.js
/**
 * @param {number[]} nums
 * @param {number} key
 * @param {number} k
 * @return {number[]}
 */
var findKDistantIndices = function (nums, key, k) {
  let keyPositions = [];
  let ans = [];
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] == key) keyPositions.push(i);
  }
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < keyPositions.length; j++) {
      const diff = Math.abs(i - keyPositions[j]);
      if (diff <= k) {
        ans.push(i);
        break;
      }
    }
  }
  return ans;
};

File: 2200_FindAllK-DistantIndicesinanArray/2200_FindAllK-DistantIndicesinanArray.py
class Solution:
    def findKDistantIndices(self, nums: List[int], key: int, k: int) -> List[int]:
        ans = []
        pos = []
        for i in range(len(nums)):
            if nums[i] == key:
                pos.append(i)
        for i in range(len(nums)): 
            for j in range(len(pos)):
                diff = abs(i - pos[j])
                if diff <= k:
                    ans.append(i)
                    break
        return ans